implement alpaka::meta::isList, alpaka::meta::ToList and alpaka::meta::toTuple

SimeonEhrig opened the pull request:
- `alpaka::meta::isTuple`: checks if a given type is a `std::tuple` or not
- `alpaka::meta::toTuple`: pack a arbitrary number of types in a `std::tuple`. If the given type is a `std::tuple` the created type is the same.

- [x] check if it working with PR #2180 , because the function is used there

psychocoderHPC commented:
@SimeonEhrig please update the description of this PR, `isTuple`,... is not valid anymore.

psychocoderHPC commented on include/alpaka/meta/TypeListOps.hpp:
```diff
@@ -35,4 +36,60 @@ namespace alpaka::meta
+    //! \brief Takes an arbitrary number of types (T) and creates a type list of type TListType with the types (T). If
+    //! T is a single template parameter and it satisfies alpaka::meta::isList, the type of the structure is T (no type
+    //! change).
+    //! \tparam TListType type of the created list
```
maybe add `e.g. std::tuple` as an example

psychocoderHPC commented on include/alpaka/meta/TypeListOps.hpp:
```diff
@@ -35,4 +36,60 @@ namespace alpaka::meta
+    //! \brief If T is a single argument and a type list (fullfil alpaka::meta::isList), the return type is T.
+    //! Otherwise, std::tuple is returned with T types as template parameters.
+    template<typename... T>
+    using toTuple = typename ToList<std::tuple, T...>::type;
```
Must be `ToTuple` because it creates the type only not the instance.

psychocoderHPC reviewed (changes requested):
ONly one name change is required and maybe a small comment extension.

SimeonEhrig committed:
11;rgb:1818/1818/1818# This is the 1st commit message:

implement alpaka::meta::isList and alpaka::meta::toList

- `alpaka::meta::isList` checks a given type is a type list
- `alpaka::meta::toList` checks if a given type is a type list and if not, create a type list with all given types
